src/components/Card/Card.js

```diff
@@ -15,6 +15,16 @@ function Card(props) {
   const [editedTitle, setEditedTitle] = useState(props.title);
   const [editedDescr, setEditedDescr] = useState(props.descr);
 
+  // Функция для сброса режима редактирования
+  const resetEditingMode = () => {
+    setIsEditing(false);
+  };
+
+  // Проверка на изменение props.readOnly и сброс режима редактирования при необходимости
+  if (props.readOnly && isEditing) {
+    resetEditingMode();
+  }
```

Comment:
I found a bug in this component.
When you start editing a card, change some text without saving or cancelling, then switch on and off read-only checkbox and then starting to change the same card again - you see your previous not saved changes.

Reply:
Fixed